### PR TITLE
fix: map branch ID in "allowedBranches" field if allowEnvTarget == true

### PR DESCRIPTION
**Pull request by michaljurecko**
Jira: https://keboola.atlassian.net/browse/PSGO-566
Slack: https://keboolaglobal.slack.com/archives/C06RG5D7R8V/p1713883324730789?thread_ts=1712838092.413209&cid=C06RG5D7R8V

**Changes:**
- Added missing mapping for branch ID, stored in `allowedBranches` manifest field.

-----------

### CODE REVIEW

File: internal/pkg/project/manifest/manifest.go [hunk 1]
```diff
@@ -110,13 +110,24 @@ func Load(ctx context.Context, logger log.Logger, fs filesystem.Fs, envs env.Pro
...
+					// Map allowed branches filter
+					mapping = append(mapping, mappingItem{
+						ManifestValue: model.AllowedBranch(originalBranchID.String()),
+						MemoryValue:   model.AllowedBranch(replacedBranchID.String()),
+					})
+				}
+				// Replace main branch in the filter, with branch ID, if needed
+				if len(content.AllowedBranches) == 1 && content.AllowedBranches[0] == model.MainBranchDef {
+					content.AllowedBranches[0] = model.AllowedBranch(replacedBranchID.String())
 				}
```
Comment (michaljurecko) [inline internal/pkg/project/manifest/manifest.go:131]: Added a new mapping, see tests.

The `allowedBranches` may contains multiple definitions, ... if the value is `<int>`, it is exact ID of the branch.
But it has different type, not `BranchID`, but `AllowedBranch`, so existing mapping definition is not working for it.

File: test/cli/pull/ok-allow-env-target/in/.keboola/manifest.json [hunk 2]
```diff
@@ -18,7 +18,7 @@
     "dataAppConfig": "app/{component_id}/{config_name}"
   },
   "allowedBranches": [
-    "__main__"
+    "2"
   ],
   "ignoredComponents": [],
   "templates": {
```
Comment (michaljurecko) [inline test/cli/pull/ok-allow-env-target/in/.keboola/manifest.json:24]: Improved tests, `2` is ID of a branch. It should be adhoc replaced by the value from ENV, ... it works already works for other occurences of the `BranchID(2)`, and now also for the `AllowedBranch("2")` definition.

File: test/cli/push/ok-allow-env-target/out/.keboola/manifest.json [hunk 3]
```diff
@@ -33,13 +33,13 @@
   },
   "branches": [
     {
-      "id": 1,
+      "id": 2,
       "path": "main"
```
Comment (michaljurecko) [inline test/cli/push/ok-allow-env-target/out/.keboola/manifest.json:37]: `1` -> `2` , because `1` is ProjectID, so test is now more clear.

**Review by hosekpeter — APPROVED** [review]
LGTM

File: internal/pkg/project/manifest/manifest.go [hunk 4]
```diff
@@ -189,6 +189,11 @@ func (m *Manifest) Save(ctx context.Context, fs filesystem.Fs) error {
 		}).(*file)
 	}
 
+	// Replace main branch in the filter, with branch ID, if needed
+	if content.AllowTargetENV && len(content.AllowedBranches) == 1 && content.AllowedBranches[0] == model.MainBranchDef {
+		content.AllowedBranches[0] = model.AllowedBranch(content.Branches[0].ID.String())
+	}
+
```
Comment (michaljurecko) [inline internal/pkg/project/manifest/manifest.go:196]: Added missing part, to the manifest `save` + updated related unit test.

File: internal/pkg/project/manifest/manifest.go [hunk 5]
```diff
@@ -189,6 +189,11 @@ func (m *Manifest) Save(ctx context.Context, fs filesystem.Fs) error {
 		}).(*file)
 	}
 
+	// Replace main branch in the filter, with branch ID, if needed
+	if content.AllowTargetENV && len(content.AllowedBranches) == 1 && content.AllowedBranches[0] == model.MainBranchDef && len(content.Branches) == 1 {
+		content.AllowedBranches[0] = model.AllowedBranch(content.Branches[0].ID.String())
+	}
+
```
Comment (michaljurecko) [inline internal/pkg/project/manifest/manifest.go:196]: I added one more missing code, see test bellow.

File: .github/workflows/test-e2e-cli.yml [hunk 6]
```diff
@@ -11,6 +11,8 @@ env:
   GO_VERSION: 1.22.2
   TEST_PARALLELISM: 8
   TEST_PARALLELISM_PKG: 1
+  # Temporary removed project: https://keboolaglobal.slack.com/archives/C055HSMKX51/p1714024266416349
+  # {"host":"connection.north-europe.azure.keboola.com","project":10892,"stagingStorage":"abs","backend":"snowflake","token":"${{ secrets.TEST_KBC_PROJECT_10892_TOKEN }}", "legacyTransformation": true}
```
Comment (michaljurecko) [inline .github/workflows/test-e2e-cli.yml:15]: There is an unrelated issue with the project: https://keboolaglobal.slack.com/archives/C055HSMKX51/p1714024266416349

**Pull request merged**